### PR TITLE
TL-287

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket

[Jira Ticket](https://hackney.atlassian.net/browse/TL-287)
## Describe this PR

Delete Responsibility Entity Implemented with tests 

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: PatchesApi.Tests/V1/Boundary/Validation/DeleteResponsibilityFromPatchRequestValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            var query = new DeleteResponsibilityFromPatchRequest() { Id = Guid.Empty };
+            var result = _classUnderTest.TestValidate(query);
+            result.ShouldHaveValidationErrorFor(x => x.Id);
+        }
+
+        [Fact]
+        public void RequestShouldErrorWithNullResponsibilityId()
+        {
+            var query = new DeleteResponsibilityFromPatchRequest();
```
Comment (evak6979) [inline PatchesApi.Tests/V1/Boundary/Validation/DeleteResponsibilityFromPatchRequestValidatorTests.cs:40]: Explicitly set null Ids

File: PatchesApi.Tests/V1/Boundary/Validation/GetByIdRequestValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        public GetByIdRequestValidatorTests()
+        {
+            _classUnderTest = new GetByIdRequestValidator();
+        }
+
+        [Fact]
+        public void RequestShouldErrorWithNullId()
+        {
+            var query = new PatchesQueryObject();
```
Comment (evak6979) [inline PatchesApi.Tests/V1/Boundary/Validation/GetByIdRequestValidatorTests.cs:24]: Explicitly set null Ids

**Review by evak6979 — APPROVED** [review]

File: PatchesApi.Tests/V1/Boundary/Validation/DeleteResponsibilityFromPatchRequestValidatorTests.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+        public DeleteResponsibilityFromPatchRequestValidatorTests()
+        {
+            _classUnderTest = new DeleteResponsibilityFromPatchRequestValidator();
+        }
+
+        [Fact]
+        public void RequestShouldErrorWithNullId()
+        {
+            var query = new DeleteResponsibilityFromPatchRequest();
```
[the remark this answers is not in the document]
Comment (humulla) [reply PatchesApi.Tests/V1/Boundary/Validation/DeleteResponsibilityFromPatchRequestValidatorTests.cs:24]: I don't think we can do that as Id is a Guid value so I can't set it to null

File: PatchesApi.Tests/V1/Boundary/Validation/GetByIdRequestValidatorTests.cs (hunk 2 above)
Comment (humulla) [reply PatchesApi.Tests/V1/Boundary/Validation/GetByIdRequestValidatorTests.cs:24]: I don't think we can do that as Id is a Guid value so I can't set it to null

File: PatchesApi.Tests/V1/Boundary/Validation/UpdatePatchQueryValidatorTests.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            var query = new UpdatePatchesResponsibilityRequest() { Id = Guid.Empty };
+            var result = _classUnderTest.TestValidate(query);
+            result.ShouldHaveValidationErrorFor(x => x.Id);
+        }
+
+        [Fact]
+        public void RequestShouldErrorWithNullResponsibilityId()
+        {
+            var query = new UpdatePatchesResponsibilityRequest();
```
[the remark this answers is not in the document]
Comment (humulla) [reply PatchesApi.Tests/V1/Boundary/Validation/UpdatePatchQueryValidatorTests.cs:40]: I don't think we can do that as Id is a Guid value so I can't set it to null

File: PatchesApi.Tests/V1/Controllers/PatchesApiControllerTests.cs [hunk 5]
```diff
@@ -91,5 +94,58 @@ public void GetPatchByIdExceptionIsThrown()
             func.Should().Throw<ApplicationException>().WithMessage(exception.Message);
         }
 
+        [Fact]
+        public async Task DeleteResponsibilityFromPatchDoesntExistReturnsNotFound()
+        {
+            // Arrange
```
[the remark this answers is not in the document]
Comment (humulla) [reply PatchesApi.Tests/V1/Controllers/PatchesApiControllerTests.cs:100]: Done in previous PR

**Pull request merged**